clang/test/SemaSYCL/non-std-layout-param.cpp

```diff
@@ -26,3 +26,16 @@ void test() {
   kernel_single_task<class MyKernel>([=] { (void)C0.Y; });
 }
 
+void test_capture_single_ref() {
+  // expected-error@+1 {{kernel parameter has non-standard layout class/struct type}}
+  int p = 0;
+  kernel_single_task<class kernel_capture_single_ref>([&p] { (void) p; });
+}
+
+void test_capture_refs() {
+  // Only one error as we stop processing after first non-standard layout parameter
```

Comment:
Why do we stop?

Reply:
Looks it's better to go though all the parameters to emit a better diagnostic, update the code to do that.

Reply:
> Looks it's better to go though all the parameters to emit a better diagnostic.

Right.